likecoin/js/admin-settings/src/components/PublishSetting.js

```diff
@@ -1,142 +1,41 @@
-            if (e.message.indexOf('password') > 0) {
-              const passwordIndex = e.message.search('password');
-              errorMessage = errorMessage.concat(
-                e.message.slice(0, passwordIndex).concat('password: "***"}'),
-              );
-            } else {
-              errorMessage = errorMessage.concat(e.message);
-            }
-          });
-        }
-        setMattersLoginError(errorMessage);
-        return;
-      }
-      const siteMattersUser = {
-        mattersId: mattersLoginResponse.viewer.userName,
-        accessToken: mattersLoginResponse.userLogin.token,
-      };
-
-      // change global state
-      updateSiteMattersLoginGlobalState(siteMattersUser);
-      setMattersLoginError(__('Success', 'likecoin'));
-      // change local state
-      setSiteMattersId(siteMattersUser.mattersId);
-    } catch (error) {
-      console.error(error);
-    }
-  }
-
-  async function loginHandler(e) {
-    e.preventDefault();
-    const mattersId = mattersIdRef.current.value;
-    const mattersPassword = mattersPasswordRef.current.value;
-    const data = {
-      mattersId,
-      mattersPassword,
-    };
-    // send to Matters API.
-    await loginToMattersAndSaveDataToWordpress(data);
-  }
-  async function handleMattersLogout(e) {
-    e.preventDefault();
-
-    // set local state
-    setSiteMattersId('');
 
-    const siteMattersUser = {
-      mattersId: '',
-      accessToken: '',
-    };
-    // change DB
-    await siteMattersLogout();
-    // change global state
-    updateSiteMattersLoginGlobalState(siteMattersUser);
-  }
-
-  async function confirmHandler() {
-    if (!showMatters) return;
-    const isSiteMattersAutoSaveDraft = siteMattersAutoSaveDraftRef.current.checked;
-    const isSiteMattersAutoPublish = siteMattersAutoPublishRef.current.checked;
-    const isSiteMattersAddFooterLink = siteMattersAddFooterLinkRef.current.checked;
+  const mattersSettingRef = useRef();
 
-    const data = {
-      siteMattersAutoSaveDraft: isSiteMattersAutoSaveDraft,
-      siteMattersAutoPublish: isSiteMattersAutoPublish,
-      siteMattersAddFooterLink: isSiteMattersAddFooterLink,
-    };
-    postSitePublishOptions(data);
-  }
+  const [showMatters, setShowMatters] = useState(!!(DBSiteMattersId
+    || DBSiteMattersAutoSaveDraft || DBSiteMattersAutoPublish));
 
   useEffect(() => {
     setShowMatters(DBSiteMattersId || DBSiteMattersAutoSaveDraft || DBSiteMattersAutoPublish);
-    setSiteMattersId(DBSiteMattersId);
-    setSiteMattersAutoSaveDraft(DBSiteMattersAutoSaveDraft);
-    setSiteMattersAutoPublish(DBSiteMattersAutoPublish);
-    setSiteMattersAddFooterLink(DBSiteMattersAddFooterLink);
   }, [
     DBSiteMattersId,
     DBSiteMattersAutoSaveDraft,
     DBSiteMattersAutoPublish,
-    DBSiteMattersAddFooterLink,
   ]);
 
+  async function confirmHandler() {
+    if (showMatters) {
+      await Promise.all([
+        mattersSettingRef.current.submit(),
+      ]);
```

Comment:
Why using `Promise.all`?

Reply:
prepare to add internet archive in next commit (not sure why github didn't show this as outdated)